firefox-ios/Client/Frontend/Settings/Main/Support/SendDataSetting.swift

```diff
@@ -84,58 +54,25 @@ class SendDataSetting: BoolSetting {
             attributedStatusText: statusText
         )
 
-        setupSettingDidChange(for: sendDataType)
+        setupSettingDidChange()
 
         // We make sure to set this on initialization, in case the setting is turned off
         // in which case, we would to make sure that users are opted out of experiments
         Experiments.setTelemetrySetting(prefs.boolForKey(prefKey) ?? true)
     }
 
-    private func setupSettingDidChange(for sendDataType: SendDataType) {
+    private func setupSettingDidChange() {
         self.settingDidChange = { [weak self] value in
-            if sendDataType != .crashReports {
-                // AdjustHelper.setEnabled($0)
-                DefaultGleanWrapper.shared.setUpload(isEnabled: value)
-
-                if !value {
-                    self?.prefs?.removeObjectForKey(PrefsKeys.Usage.profileId)
-
-                    // set dummy uuid to make sure the previous one is deleted
-                    if let uuid = UUID(uuidString: "beefbeef-beef-beef-beef-beeefbeefbee") {
-                        GleanMetrics.Usage.profileId.set(uuid)
-                    }
-                }
-
-                Experiments.setTelemetrySetting(value)
-                self?.shouldSendData?(value)
-            }
+            self?.shouldSendData?(value)
```

Comment:
Now that this code is removed, do we need  `shouldSendData`?

Reply:
Yes, we are using it AppSettingsTableViewController callback for every setting.